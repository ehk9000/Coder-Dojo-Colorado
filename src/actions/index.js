export const addUser = (user) => ({
  type:'ADD_USER',
  user
});

export const isFetching = (bool) => ({
  type: 'IS_FETCHING',
  isFetching: bool
});