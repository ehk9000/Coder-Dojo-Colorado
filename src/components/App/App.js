import React, { Component } from 'react';
import queries from '../../utils/queries/queries';
import LoginForm from '../LoginForm/LoginForm';

class App extends Component {
  state = {
    error: '',
    users: []
  }

  async componentDidMount() {
    try {
      const users = await queries.getUsers();

      this.setState({ users: users.allUsers });
    } catch ({ message }) {
      this.setState({ error: message });
    }
  }
  
  createUser = async () => {
    const mockUser = {
      email: 'tiff@sum.com',
      nickname: 'ERICCCCCC',
      password: 'asdfasdf'
    };
    
    await queries.newUser(mockUser);
  }

  render() {
    return (
      <div className="App">
        <h1>CODERDOJO!</h1>
        <button type="submit" onClick={this.createUser}>
          CREATE
        </button>
        <LoginForm />
      </div>
    );
  }
}

export default App;
