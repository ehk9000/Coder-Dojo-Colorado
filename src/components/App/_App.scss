* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
}

body {
  font-family: sans-serif;
}

div.App {
  @include lato-font;
  height: 100vh;
  width: 100vw;
  background-color: #EEEEEE;
    
  h1 {
    color: red;
  }
}
