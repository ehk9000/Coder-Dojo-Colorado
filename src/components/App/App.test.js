import React from 'react';
import { shallow } from 'enzyme';
import { App, mapDispatchToProps } from './App';
import * as actions from '../../actions';

jest.mock('../../actions');

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App setUser={actions.addUser}/>);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mdtp', () => {
    it('should dispatch setUser', () => {
      const mockUser = {
        id: 1,
        username: 'tiff',
        password: 'password'
      };
      const action = actions.addUser(mockUser);
      const mockDispatch = jest.fn();
      
      mapDispatchToProps(mockDispatch).setUser();

      expect(mockDispatch).toHaveBeenCalledWith(action);
    });
  });
});
