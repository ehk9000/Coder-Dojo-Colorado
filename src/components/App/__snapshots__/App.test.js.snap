// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="App"
>
  <h1>
    CODERDOJO!
  </h1>
  <button
    onClick={[Function]}
    type="submit"
  >
    CREATE
  </button>
  <GuardianDetailsForm
    handleChange={[Function]}
    handleSubmit={[Function]}
  />
  <LoginForm
    addUser={[Function]}
  />
</div>
`;
